src/components/image-import/index.js

```diff
@@ -0,0 +1,159 @@
+import { forwardRef, useCallback, useRef } from "@wordpress/element";
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import DropZone from "../../elements/drop-zone";
+import ImageInput from "../../elements/image-input";
+import Label from "../../elements/label";
+
+/**
+ * @param {string} id Id.
+ * @param {string} name Name.
+ * @param {string} imageInputVariant Image input variant.
+ * @param {string} previewImage Preview image.
+ * @param {string} previewImageAlt Preview image alt.
+ * @param {string} importText Import text.
+ * @param {string} buttonVariant Button variant.
+ * @param {Function} onClick On click callback.
+ * @param {Function} onChange On change callback.
+ * @param {Function} onReset On reset callback.
+ * @param {Function} onDrop On drop callback.
+ * @param {Function} onError On error callback.
+ * @param {string} dropLabel Drop label.
+ * @param {boolean} disabled Disabled state.
+ * @param {string} resetPreviewActionAriaLabel Reset preview action aria label.
+ * @param {string} className Classname.
+ * @param {string} children Children.
+ * @returns {JSX.Element} The ImageImport component.
+ * */
+const ImageImport = forwardRef( ( {
+	id,
+	name,
+	imageInputVariant,
+	previewImage,
+	previewImageAlt,
+	importText,
+	buttonVariant,
+	onClick,
+	onChange,
+	onReset,
+	onDrop,
+	onError,
+	dropLabel = "Select an image or drag and drop",
+	disabled,
+	resetPreviewActionAriaLabel,
+	className,
+	children,
+	...props
+}, ref ) => {
+	const inputRef = useRef();
+
+	const handleOnDrop = useCallback( ( event ) => {
+		if ( ! event.dataTransfer.files ) {
+			return;
+		}
+
+		const file = event.dataTransfer.files[ 0 ];
+		if ( file ) {
+			try {
+				// Verify that file is an image
+				if ( ! file.type.startsWith( "image/" ) ) {
```

Comment:
@wpalani SVGs will return as `image/svg+xml`. Is there somewhere else we're restricting this? As much as I want to see SVG support normalized in WordPress, I think we need an exclusion in part to assure crops, image optimization efforts and other things aren't unexpectedly impacted.

Reply:
@wpalani related, there is a way to tap the currently-supported MIME types for WordPress images.

```js
// Access the supported MIME types in WordPress.
const mimeTypes = wp.media.view.settings.mimeTypes;

// Log or use the MIME types.
console.log(mimeTypes);
```

Under the hood it's hitting `get_allowed_mime_types();`
```php
/ Get all allowed MIME types.
$mime_types = get_allowed_mime_types();

// Filter only image MIME types.
$image_mime_types = array_filter($mime_types, function ($mime, $ext) {
    return strpos($mime, 'image/') === 0; // Keep only image types.
}, ARRAY_FILTER_USE_BOTH);
```

Reply:
Just to stay pure to the library's goal, and since the component is an image input, I think it should accept all image types, then pass the event to the consuming app and then the consumer can do what they want with types?

the drop event is already being passed up here: https://github.com/newfold-labs/npm-ui-component-library/pull/22/files#diff-6793a61a956e59d0d38519d55506916a3ead3004c7fe2a905fd3735038787409R84